pkg/logs/pipeline/pipeline.go

```diff
@@ -20,24 +20,39 @@ type Pipeline struct {
 }
 
 // NewPipeline returns a new Pipeline
-func NewPipeline(connManager *sender.ConnectionManager, outputChan chan message.Message) *Pipeline {
-
+func NewPipeline(outputChan chan message.Message, endpoints *config.Endpoints) *Pipeline {
 	useProto := config.LogsAgent.GetBool("logs_config.dev_mode_use_proto")
+	useSSL := endpoints.Main.UseSSL
+
+	// initialize the main destination
+	main := sender.NewDestination(
+		sender.NewAPIKeyPrefixer(endpoints.Main.APIKey, endpoints.Main.Logset),
+		sender.NewDelimiter(useProto),
+		sender.NewConnectionManager(endpoints.Main.Host, endpoints.Main.Port, useSSL, endpoints.Main.ProxyAddress),
+	)
+
+	// initialize the additional destinations
+	var additionals []*sender.Destination
+	for _, endpoint := range endpoints.Additionals {
+		additionals = append(additionals, sender.NewDestination(
```

Comment:
looks like the `NewDestination()` from and endpoint could be factorized (it looks a lot like the one above)